### ISSUE TITLE
What is the right way to use ChromeManifest

### ISSUE THREAD

**Issue by onemen** [onemen/TabMixPlus#112]
@117649,

In our bootstrap startup we set `chromeManifest` one time for all documents calls from `chrome-document-loaded`, but for the case `reason === ADDON_INSTALL || (reason === ADDON_ENABLE && !window.Tabmix)` we create new `chromeManifest` for each window.

What is the right way?

https://github.com/onemen/TabMixPlus/blob/f1e825e1bc9426b11a0cce104606f15c7477090d/addon/bootstrap.js#L136-L143

https://github.com/onemen/TabMixPlus/blob/f1e825e1bc9426b11a0cce104606f15c7477090d/addon/bootstrap.js#L168-L172

**Comment by 117649** [comment]
I think they are same.
The code in `ADDON_ENABLE` is for the situation that addon is start after browser startup and to catch browser windows already opened before addon is enabled. Given the situation the two `chromeManifest` could be different base on the need but in our current case its should be same.

As far as I can remember we don't put 'overlay' segments back into `chrome.manifest` file and read the file is because Firefox would complain if it saw those there.

**Comment by onemen** [comment]
If i understand it correctly we can use one `chromeManifest` for the entire startup function.
we can move
```js
   const chromeManifest = new ChromeManifest(() => { 
     return man; 
   }, options); 
```
to the beginning of startup function.

Regarding `chrome.manifest` file, i did not see any place in `ChromeManifest.jsm` that read it.
The only place that read it is in @xiaoxiaoflood in [firefox-scripts utils/boot.jsm](https://github.com/xiaoxiaoflood/firefox-scripts/blob/master/chrome/utils/boot.jsm)

```js
let cmanifest = Cc['@mozilla.org/file/directory_service;1'].getService(Ci.nsIProperties).get('UChrm', Ci.nsIFile);
cmanifest.append('utils');
cmanifest.append('chrome.manifest');
Components.manager.QueryInterface(Ci.nsIComponentRegistrar).autoRegister(cmanifest);

ChromeUtils.import('chrome://userchromejs/content/BootstrapLoader.jsm');

ChromeUtils.import('chrome://userchromejs/content/userChrome.jsm');
```

**Comment by 117649** [comment]
yes

**Comment by onemen** [comment]
fixed by c508d162178e6fb3d2f1d53ee2e704b913884e65
